Gems/Atom/RHI/Code/Source/RHI/ShaderResourceGroupData.cpp

```diff
@@ -102,7 +102,11 @@ namespace AZ::RHI
 
                 for (int imageIndex = 0; imageIndex < imageViews.size(); ++imageIndex)
                 {
-                    deviceImageViews[imageIndex] = imageViews[imageIndex] ? imageViews[imageIndex]->GetDeviceImageView(deviceIndex).get() : nullptr;
+                    deviceImageViews[imageIndex] = imageViews[imageIndex]
+                        ? (int(imageViews[imageIndex]->GetImage()->GetDeviceMask()) & 1 << deviceIndex
```

Comment:
Can we move this line to a method as it is used below? Use c++ cast.

Reply:
Done.